react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+class Drawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isOpen: false };
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
+    if (this.state.isOpen) {
+      this.setState({ isOpen: false });
+    } else {
+      this.setState({ isOpen: true });
+    }
+  }
+
+  render() {
+    let button;
+    if (this.state.isOpen) {
+
+      button = <div>
+      <div className="">
+
+          <i className="fas fa-bars" onClick={this.toggle}></i>
+
+        </div>
+        <div className="nav-div on-screen">
+          <h2 onClick={this.toggle}>Menu</h2>
+          <div>
+          <ul>
+              <li><p onClick={this.toggle}>About</p></li>
+              <li><p onClick={this.toggle}>Get Started</p></li>
+              <li><p onClick={this.toggle}>Sign In</p></li>
+          </ul>
+
+         </div>
+      </div>
+      </div>;
+    } else if (!this.state.isOpen) {
+      button = <div>
+       <div>
+
+        <i className="fas fa-bars" onClick={this.toggle}></i>
+
+      </div>
+        <div className="nav-div off-screen" >
+          <h2 onClick={this.toggle}>Menu</h2>
+        <div>
+          <ul>
+            <li><p onClick={this.toggle}>About</p></li>
+            <li><p onClick={this.toggle}>Get Started</p></li>
+            <li><p onClick={this.toggle}>Sign In</p></li>
+          </ul>
+
+         </div>;
+
+        </div>
+      </div>;
+
+    }
```

Comment:
It looks like the only different between these two element trees is the `on-screen` and `off-screen` classes. Try to refactor your code so that it conditionally decides between those two classes instead of the entire element tree.